docs/contributor/arch/016-transform-filter-support-for-telemetry-pipelines.md

```diff
@@ -0,0 +1,118 @@
+
+Example configuration MetricPipeline:
+    
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: MetricPipeline
+metadata:
+  name: metricpipeline-sample
+spec:
+  transform:
+    conditions:
+      - type == METRIC_DATA_TYPE_SUM
+    statements:
+      - set(description, "Sum")
+  filter:
+    metric:
+        - type == METRIC_DATA_TYPE_NONE
+    datapoint:
+        - metric.name == "k8s.pod.phase" and value_int == 4
+  input:
+      istio:
+        enabled: true
+      prometheus:
+        enabled: false
+  output:
+    otlp:
+      endpoint:
+        value: ingest-otlp.services.sap.hana.ondemand.com:443
+```
+
+Example configuration TracePipeline:
+
+```yaml
+apiVersion: telemetry.kyma-project.io/v1alpha1
+kind: TracePipeline
+metadata:
+  name: tracepipeline-sample
+spec:
+  transform:
+    statements:
+        - set(status.code, 1) where attributes["http.path"] == "/health"
+  filter:
+    span:
+      - IsMatch(resource.attributes["k8s.pod.name"], "my-pod-name.*")
+  output:
+    otlp:
+      endpoint:
+        value: ingest-otlp.services.sap.hana.ondemand.com:443
+      protocol: grpc
+```
+### Solution
+
+The OTTL library offers functions to validate, filter, and transform data by parsing and executing expressions. To ensure robustness and stability, we will limit the available filter and transform functions to a carefully selected subset. This subset will be curated based on common use cases, ensuring essential functionality without overcomplicating the configuration process, the OTTL library provides two categories of functions, the [editor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#editors) functions and the [converters](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/ottlfuncs#converters).
+The editor functions are used to modify the data, while the converters are used in the OTTL condition expressions to convert the data type to be able to compare it with the expected value. The current existing pipelines use a small subset of OTTL functions like setting a value, adding or removing attributes, and filtering data based on conditions. The solution will use this subset of editor functions to provide the necessary functionality to the users, the additional functions when required can be added.
+The converter functions support a broad variety of data types, including xlm, json, and string; not all of these functions are required for the current use cases, basic operation like string comparison, formatting and data type checks could be sufficient. 
```

Comment:
```suggestion
The converter functions support a broad variety of data types, including xml, json, and string; not all of these functions are required for the current use cases, basic operation like string comparison, formatting and data type checks could be sufficient. 
```
sorry, it took me too long to finally figure out what looked wrong all the time :)